config/config.go

```diff
@@ -0,0 +1,77 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/pelletier/go-toml"
+	"github.com/pkg/errors"
+)
+
+const menmosConfigDirName = "menmos"
+const menmosConfigFileName = "client.toml"
+
+// A Config represents the on-disk configuration of a menmos client.
+type Config struct {
+	Profiles map[string]Profile `json:"profiles,omitempty"`
+}
+
+func loadConfigFromFile(path string) (*Config, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to open menmos configuration file")
+	}
+
+	decoder := toml.NewDecoder(file).SetTagName("json")
+
+	var cfg Config
+	err = decoder.Decode(&cfg)
+
+	if err != nil {
+		err = errors.Wrap(err, "failed to decode TOML config")
+	}
+
+	return &cfg, err
+}
+
+func getDefaultConfigPath() (string, error) {
+	configPath, err := os.UserConfigDir()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get the user configuration directory")
+	}
+
+	menmosConfigDirPath := path.Join(configPath, menmosConfigDirName)
+
+	if err := os.MkdirAll(menmosConfigDirPath, 0644); err != nil {
```

Comment:
Not a fan of the argument in this case because Go doesn't have default arguments. I agree it needs to be made explicit though, so I'll change `LoadDefault` to `LoadOrCreateDefault`, and move the creation logic from `getDefaultConfigPath` to `LoadOrCreateDefault`